### ISSUE TITLE
Optical source missing after 20201213 update

### ISSUE THREAD

**Issue by laurillev** [hifiberry/hifiberry-os#197]
**Describe the bug**
Optical source is not visible anymore in Sources and no sound is produced from toslink input.

**HiFiBerryOS version**
20201213

**HiFiBerry sound card**
DAC+DSP

**To Reproduce**
1. Go to Sources-tab in Hifiberry OS
2. Look through the list

**Expected behavior**
Expected to have Optical source as option and sound to be produced as before.

**Browser (if applicable)**

**Additional context**
DSP program is “DAC+ DSP Universal Version 12”

**Comment by LarsenDX** [comment]
Hi,
Same here. Optical source missing since 20201213.

**Comment by ndamkjaer** [comment]
Hi,
I also have this issue. 
Running Pi 4 with DAC+ DSP.

**Comment by TheWaveHunter** [comment]
Hi,
Same here. Using optical input as primary source.
Pi 4, after update from 20201101 to 20201213

**Comment by hifiberry** [comment]
Ok, this is a bug in the upgrade process. Try the following

rm /etc/hifiberry.state
/opt/hifiberry/bin/reconfigure-players

**Comment by laurillev** [comment]
Unfortunately that didn't solve it for me.

Mine is a new installation, so it shouldn't have to do with upgrade. But thus I cannot temporarily go back to previous version either.

**Comment by hifiberry** [comment]
What's the output of the reconfigure-players command?

**Comment by laurillev** [comment]
cat: /tmp/eeprom.txt: No such file or directory
cat: /tmp/eeprom.txt: No such file or directory
/etc/hifiberry ~
~
pi
bluetooth
pi3orlater
arm7
dsp
localui
DAC+ DSP 1
Nothing changed, exiting...

**Comment by hifiberry** [comment]
Hmm, this looks different than what it should be. What's the output of

grep toslink /opt/hifiberry/bin/reconfigure-players

**Comment by laurillev** [comment]
grep toslink /opt/hifiberry/bin/reconfigure-players 
     FEATURES="$FEATURES spdifnoclock toslink"
   FEATURES="$FEATURES toslink"
      FEATURES="$FEATURES dsp toslink"

**Comment by hifiberry** [comment]
Ok, that also looks right. I need to have another look.

**Comment by laurillev** [comment]
Maybe that toslink "feature" is not correctly set for DAC+DSP somewhere.

**Comment by hifiberry** [comment]
Yes, that's right. Question is just WHY

**Comment by hifiberry** [comment]
Ok, looks like I found the problem. We'll fix it in the next release. For now, just use a workaround:

touch/etc/hifiberry/toslink.feature
/opt/hifiberry/bin/reconfigure-players

**Comment by schwachstelle** [comment]
Looks like a typo; in which directory should the feature file be placed?

**Comment by laurillev** [comment]
@schwachstelle use space between touch-command and '/etc/...'

**Comment by schwachstelle** [comment]
@laurillev yes, but there is no directory /etc/hifiberry. Maybe /opt/hifiberry/etc ?

**Comment by hifiberry** [comment]
Just create the directory

**Comment by zaneclaes** [comment]
Just confirming that the following two commands, alone, worked for me with my Beocreate (note there was a typo with a slash after `touch` in the original):

```
touch etc/hifiberry/toslink.feature
/opt/hifiberry/bin/reconfigure-players
```

**Comment by David91919** [comment]
I have the same issue and needed to make a fresh install so I can´t go back to the old version.

What are the exact commands for the fix ?

Which directory do I need to create?

Running these commands just gets me errors:
touch etc/hifiberry/toslink.feature          > touch: cannot touch 'etc/hifiberry/toslink.feature': No such file or directory

/opt/hifiberry/bin/reconfigure-players
```
# /opt/hifiberry/bin/reconfigure-players
cat: /tmp/eeprom.txt: No such file or directory
cat: /tmp/eeprom.txt: No such file or directory
/etc/hifiberry ~
~
pi
bluetooth
pi3orlater
arm7
dsp
localui
Beocreate 4-Channel Amplifier 1
Nothing changed, exiting...
```

**Comment by hifiberry** [comment]
See https://github.com/hifiberry/hifiberry-os/issues/197#issuecomment-748946792

**Comment by David91919** [comment]
> See [#197 (comment)](https://github.com/hifiberry/hifiberry-os/issues/197#issuecomment-748946792)

its not working :-( fresh install after download from website, root login

```
# touch/etc/hifiberry/toslink.feature
-bash: touch/etc/hifiberry/toslink.feature: No such file or directory
# /opt/hifiberry/bin/reconfigure-players
cat: /tmp/eeprom.txt: No such file or directory
cat: /tmp/eeprom.txt: No such file or directory
/etc/hifiberry ~
~
pi
bluetooth
pi3orlater
arm7
dsp
localui
Beocreate 4-Channel Amplifier 1
Nothing changed, exiting...
```

**Comment by hifiberry** [comment]
Just put a space between touch and the filename

**Comment by philaub** [comment]
> Ok, looks like I found the problem. We'll fix it in the next release. For now, just use a workaround:
> 
> touch/etc/hifiberry/toslink.feature /opt/hifiberry/bin/reconfigure-players

Are there any updates? The "next release" batch seems a bit odd now I think.
